titanium/app/controllers/search.js

```diff
@@ -0,0 +1,186 @@
+// Create/Reference instances of all collections we need
+var properties = Alloy.Collections.instance('property');
+var favorites = Alloy.Collections.instance('favorite');
+var searches = Alloy.Collections.instance('search');
+
+// Self-executing constructor, just for clarity
+(function constructor(args) {
+
+  // Fetch favorites from SQL data source so we can use them in different controllers
+  favorites.fetch();
+
+  // Show/Hide the ambigious/recent locatins when collection updates and do so now
+  searches.on('fetch reset', showOptions);
```

Comment:
@FokkeZB you set `on` events, but never remove them via `off` in this controller. Is that intentionally or an oversight?